### DISCUSSION TITLE
Setup lldap with Synology DSM

### DISCUSSION THREAD

**Discussion by Karstensson** [lldap/lldap#737] [Q&A]
Helo, Have anyone managed to get lldap working with synology DSM? These are the fields to fill in. I manage to get users identified but run in to issues with groups.

Without fully understanding what I'm doing, i have tried to copy as much as possible from other example configs.

![image](https://github.com/lldap/lldap/assets/62674282/3c8a098a-0c3e-4b96-b6ce-da209e41245a)

**Comment by nitnelave** [comment]
Hi! Sadly, last I checked, the design of Synology and LLDAP are incompatible: Synology wants to have the user's password hash to check themselves if the password is correct, but LLDAP doesn't store that. Instead, LLDAP stores enough information to do a zero knowledge proof that the password is correct, using the OPAQUE protocol.

The issue came up before, and we couldn't find a way forward. If you manage to find a way to make Synology work without the userPassword field, others would be interested.

For context, what every other service does is that they try to login into LLDAP with the user's credentials: if it succeeds, the password is correct.

**Reply by Karstensson** [reply]
Oh okay I understand. Thank for the clarification!

**Comment by gon3ales** [comment]
<img width="481" height="498" alt="synology" src="https://github.com/user-attachments/assets/fee5c228-db20-415f-8888-c9ff52a03a9e" />
Try this scheme
